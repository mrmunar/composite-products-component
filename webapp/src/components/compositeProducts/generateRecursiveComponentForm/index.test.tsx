import React from 'react';
import { useSelector, useDispatch } from "react-redux";
import { render, cleanup } from '@testing-library/react';
import GenerateRecursiveComponentForm from './';

afterEach(cleanup);

const testData = {
    id: 'efd0dab3-4813-44f0-98bc-ea3366582498',
    name: 'Example 1',
    components: [{
        type: 'PRODUCT',
        quantity: 1,
        productId: 'ba7a0995-4193-4589-8150-54a3cef5f2c6'
    }, {
        type: 'GROUP',
        label: 'Test Group',
        components: []
    }]
};

jest.mock('react-redux', () => ({
    useDispatch: jest.fn()
}));

// Mocking ProductItem component. Will test this in isolation
jest.mock('../productItem', () => () => 'ProductItem');

const mockUseDispatch = useDispatch as jest.Mock;
const mockDispatch = jest.fn();

mockUseDispatch.mockImplementation(() => mockDispatch);

test('should take a snapshot', () => {
    const { baseElement } = render(<GenerateRecursiveComponentForm
        compositeProduct={testData}
        onChange={() => { }}
    />);
    expect(baseElement).toMatchSnapshot();
});

test('should render testData', () => {
    const { getByTestId } = render(<GenerateRecursiveComponentForm
        compositeProduct={testData}
        onChange={() => { }}
    />);
    expect(getByTestId('group-container')).toHaveTextContent('Group Label');
});