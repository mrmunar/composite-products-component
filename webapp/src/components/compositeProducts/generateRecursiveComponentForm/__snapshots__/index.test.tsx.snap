// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<body>
  <div>
    <div
      class="m-2 p-3 card"
    >
      ProductItem
      <div
        class="mt-2"
        data-testid="group-container"
      >
        <div>
          Group Label
        </div>
        <div
          class="row"
        >
          <div
            class="col-sm-8 text-left"
          >
            <input
              class="form-control"
              data-testid="input-text"
              type="text"
              value="Test Group"
            />
          </div>
          <div
            class="col-sm-4 text-right"
          >
            <button
              class="btn btn-danger"
              data-testid="button"
            >
              X
            </button>
          </div>
        </div>
        <div
          class="m-2 p-3 card"
        >
          <div
            class="mb-2 mt-2"
          >
            <select
              class="custom-select"
              data-testid="select"
            >
              <option
                value=""
              >
                (Add +)
              </option>
              <option
                value="add-product"
              >
                Add Product
              </option>
              <option
                value="add-group"
              >
                Add Group
              </option>
            </select>
          </div>
        </div>
      </div>
      <div
        class="mb-2 mt-2"
      >
        <select
          class="custom-select"
          data-testid="select"
        >
          <option
            value=""
          >
            (Add +)
          </option>
          <option
            value="add-product"
          >
            Add Product
          </option>
          <option
            value="add-group"
          >
            Add Group
          </option>
        </select>
      </div>
    </div>
  </div>
</body>
`;
